exec/java-exec/src/main/resources/drill-module.conf

```diff
@@ -91,7 +91,22 @@ drill.exec: {
     use.ip : false
   },
   optimizer: {
-    implementation: "org.apache.drill.exec.opt.IdentityOptimizer"
+    implementation: "org.apache.drill.exec.opt.IdentityOptimizer",
+    enhanced_defaults: {
```

Comment:
You may pass in a difference instance of DrillDefaultRelMetaDataProvider, based on context.getPlannerSetting(). 

https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DefaultSqlHandler.java#L382